### ISSUE TITLE
'Error loading price' might happen at any moment

### ISSUE THREAD

**Issue by elena-zh** [gnosis/cowswap#1283]
The issue is not always reproducible, hover, at some moment, error appears when Swap form is opened for too long, or when swap tokens. 
![image](https://user-images.githubusercontent.com/70885163/130061267-f27ed640-6ec6-4243-b458-98e8e4906039.png)

See the video: it is for about 5 min, I caught the error after 1 min there
https://drive.google.com/file/d/15ueFhPuFLgzxg7wZEr8yawHCobZAXiXJ/view

**Comment by elena-zh** [comment]
Sometimes, it is even may look like this 
![error loading price](https://user-images.githubusercontent.com/70885163/130254334-4cd98810-d932-4421-abf3-831413863ac5.jpg)

**Comment by MareenG** [comment]
I run into this issue quite often while testing the new RC #1425 
I noticed that this happens quite often for tokens that where not approved yet. 
So I traded DAI vs a token (on Rinkeby) that was not approved afterwards I changed the tokens to revert the swap. Instead of approve tokens most of the time I got the 'Error loading price' notification appeared
Not sure if this is a coincidence but just wanted to mention this.

<img width="1041" alt="Bildschirmfoto 2021-09-15 um 12 02 43" src="https://user-images.githubusercontent.com/34510341/133415830-b859091f-3f07-4e53-bf87-d51d18ae9c18.png">
